src/Configuration/Dotenv/Scheme/Helpers.hs

```diff
@@ -0,0 +1,26 @@
+{-# LANGUAGE RecordWildCards #-}
+
+module Configuration.Dotenv.Scheme.Helpers where
+
+import Data.List
+
+import Configuration.Dotenv.Scheme.Types
+
+matchVarWithType
+  :: Config           -- ^ List of EnvConf for variables
+  -> (String, String) -- ^ (Env Name, Env Value)
+  -> (String, EnvType)
+matchVarWithType (Config envConfs) (name, value) =
+  let criteria EnvConf{..} = name `elem` envNames
+      maybeEnvConf = find criteria envConfs
+      pairEnvWithConf EnvConf{..} = (value, envType)
+   in
+   case maybeEnvConf of
```

Comment:
Nice... this way we can catch that the user is declaring all the variables in the dotenv...